tests/framework/metrics.go

```diff
@@ -0,0 +1,145 @@
+package framework
+
+import (
+	"context"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
```

Comment:
Per a conversation offline, I bet it would be easier to get metrics if we just deploy our own Prometheus, and not use the GCP monitoring. Then we also don't need this extra dependency.